### PR TITLE
Use readable byte streams

### PR DESCRIPTION
**Pull request by domenic**
Depends on https://github.com/whatwg/streams/pull/1130. Closes https://github.com/WICG/serial/issues/127.

### CODE REVIEW

File: index.html
Comment (reillyeon) [inline index.html:790]: Can you export these symbols from the Streams specification so hard links aren't necessary?

Comment (domenic) [reply index.html:790]: They are exported; the problem is that ReSpec doesn't seem to support `for=""` values with slashes in them. /cc @marcoscaceres

Comment (marcoscaceres) [reply index.html:790]: We do support it: https://respec.org/xref/?term=pullalgorithm ... but seems there is a bug.  

I'd suggest just putting in [=ReadableStream/set up/pullAlgorithm=]. I'll investigate why it's not linking.

File: index.html
Comment (marcoscaceres) [inline index.html:792]: Same with these ones :)

File: index.html
Comment (marcoscaceres) [reply index.html:790]: Filed. https://github.com/w3c/respec/issues/3595 ... hope to have a fix soon.

Comment (marcoscaceres) [reply index.html:790]: Ok, should be be fixed in respec@26.8.10 😅🤞

Comment (marcoscaceres) [inline index.html:790]:
```suggestion
        <i>[=ReadableStream/set up/pullAlgorithm=]</i>
```

File: index.html
Comment (marcoscaceres) [reply index.html:792]: This one doesn't seem to exist?

File: index.html
Comment (marcoscaceres) [inline index.html:794]: Same with this one... not in the spec?

Comment (domenic) [reply index.html:794]: None of these are in the spec yet; this PR depends on https://github.com/whatwg/streams/pull/1130.

Comment (marcoscaceres) [reply index.html:794]: Ah, ok. Hopefully they should be all good once that's merged. Any linking/respec issues, just ping 😇

File: index.html
Comment (marcoscaceres) [inline index.html:698]: @domenic, sorry, could you change all the internal slots to be namespaced? This depends on #145. 

```suggestion
            [=this=].{{SerialPort/[[readable]]}}.
```

**Comment by domenic** [comment]
Alright, the Streams spec PR is merged, and this is rebased and should be up to date. We'll need some time for the terms to make their way into databases, but hopefully then the build should pass.

It's still not 100% clear to me whether we should merge this or wait for it to get on the near-term implementation roadmap first; that's up to the editors. But I think it's been a very valuable exercise in designing Streams's other-spec integration points so it's time well spent for me :)

**Comment by reillyeon** [comment]
I'm very excited to get this shipped but we should hold off on merging until there's at least a timeline for getting it into a Chromium release. Right now that's being tracked as [issue 1182905](https://crbug.com/1182905) and is blocked on changes to the Streams C++ bindings to support natively implemented BYOB streams.

**Review by reillyeon — APPROVED** [review]

**Comment by marcoscaceres** [comment]
@reillyeon, if you re-trigger the CI, it should be happy now 🤞

**Comment by reillyeon** [comment]
Now that Chromium supports implementing APIs that provide readable byte streams I have created an implementation of this PR which is currently under review: https://chromium-review.googlesource.com/c/chromium/src/+/3811977

**Pull request merged**
